src/shared/capture/capture_spinnaker.cpp

```diff
@@ -0,0 +1,398 @@
+//========================================================================
+//  This software is free: you can redistribute it and/or modify
+//  it under the terms of the GNU General Public License Version 3,
+//  as published by the Free Software Foundation.
+//
+//  This software is distributed in the hope that it will be useful,
+//  but WITHOUT ANY WARRANTY; without even the implied warranty of
+//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+//  GNU General Public License for more details.
+//
+//  You should have received a copy of the GNU General Public License
+//  Version 3 in the file COPYING that came with this distribution.
+//  If not, see <http://www.gnu.org/licenses/>.
+//========================================================================
+/*!
+  \file    capture_spinnaker.cpp
+  \brief   C++ Implementation: CaptureSpinnaker
+  \author  Nicolai Ommer, (C) 2019
+*/
+//========================================================================
+
+#include "capture_spinnaker.h"
+#include <memory>
+#include <iostream>
+
+#ifndef VDATA_NO_QT
+CaptureSpinnaker::CaptureSpinnaker(VarList * _settings,int default_camera_id, QObject * parent) : QObject(parent), CaptureInterface(_settings)
+#else
+CaptureSpinnaker::CaptureSpinnaker(VarList * _settings,int default_camera_id) : CaptureInterface(_settings)
+#endif
+{
+  cam_id = (unsigned int) default_camera_id;
+  is_capturing = false;
+
+  #ifndef VDATA_NO_QT
+    mutex.lock();
+  #endif
+
+  settings->addChild(capture_settings = new VarList("Capture Settings"));
```

Comment:
Please split into two statements for clarity.

Reply:
I agree that it looks nicer when splitted. On the other hand, all capture classes do it like this. I wanted to be consistent with the rest of the vision code.

Reply:
Okay, let's go for consistency.